src/prefect/agent.py

```diff
@@ -143,33 +149,43 @@ async def get_work_queues(self) -> AsyncIterator[WorkQueue]:
                 work_queue = await self.client.read_work_queue_by_name(
                     work_pool_name=self.work_pool_name, name=name
                 )
-            except ObjectNotFound:
-                # if the work queue wasn't found, create it
-                if not self.work_queue_prefix:
-                    # do not attempt to create work queues if the agent is polling for
-                    # queues using a regex
-                    try:
-                        work_queue = await self.client.create_work_queue(
-                            work_pool_name=self.work_pool_name, name=name
-                        )
-                        if self.work_pool_name:
-                            self.logger.info(
-                                f"Created work queue {name!r} in work pool"
-                                f" {self.work_pool_name!r}."
-                            )
-                        else:
-                            self.logger.info(f"Created work queue '{name}'.")
-
+            except (ObjectNotFound, Exception):
+                self.logger.exception("Failed to query work queue by name.")
```

Comment:
I don't think that we need an exception-level log here. If the queue doesn't exist then we'll either attempt to create it or log an error further down. 
```suggestion
```